collections/src/main/scala/strawman/collection/immutable/LazyList.scala

```diff
@@ -359,6 +469,11 @@ sealed private[immutable] trait LazyListOps[+A, +CC[+X] <: LinearSeq[X] with Laz
 
   override final def zipWithIndex: CC[(A, Int)] = this.zip(LazyList.from(0))
 
+  def toStringBuilder(b: StringBuilder): StringBuilder
+
+  override def toString: String = {
+    s"$className${toStringBuilder(new StringBuilder).mkString("(", ", ", ")")}"
```

Comment:
What happens if my elements take several characters? `LazyList(10).force.toString == LazyList(1, 0)`?

Reply:
I think we shouldn’t have to use `mkString`: we already have a `StringBuilder` on hand, we should manage to add the parens and commas when we add the elements to it.

Reply:
Removed an extra call to `mkString`. Added logic of adding seperator in `toStringBuilder` method .